### PR TITLE
Group10

### PR DESCRIPTION
**Pull request by devmsh**

### CODE REVIEW

File: ShippingTest.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     * User must be able to enter the shipping address
+     * */
+    public function testUserMustBeAbleToEnterShippingAddress(){
+
+
+        $client = new Client();
+        $client->setName("Ahmed");
+        $client->setEmail("ahmad@email.com");
+        $address = new Address("115","Al-Nasser St.","Gaza","Palestime");
+        $this->assertTrue($client->setAddress($address));
```
Comment (devmsh) [inline ShippingTest.php]: Assert `true` is not a good approach, and it did not indicate that we really set the `address`.

Use `getAddress` and make sure that the returned object is the same as the object you send to the setter method.

File: ShippingTest.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        // Create a ship Service Class.
+        $ship = $this->createMock(ShippingService::class);
+
+        // Configure the ship.
+        $ship->method('isOrderShipped')->willReturn(true);
+
+        $order = Order::find(123456);
+
+        $this->assertTrue($order->checkShippingStatus($ship));
+        $this->assertEquals("Shipped",$order->getStatus());
```
Comment (devmsh) [inline ShippingTest.php]: Here, we test to `isOrderShipped` mock when it return `true`. We also need to test it again and return mocked `false`

Comment (anasanwar) [inline ShippingTest.php]: I used (function testFailureShippingOrderStatusStillShipingAndNoMessageSent) check it.
is it correct??

File: ShippingTest.php (hunk 1 above)
Comment (AbdelRahmanHamada) [inline ShippingTest.php]: OK, this will be:
$this->setAddress($address);
$this->assertEqual($address, $client->getAddress());

Comment (devmsh) [inline ShippingTest.php]: :+1:

File: ShippingTest.php (hunk 2 above)
Comment (devmsh) [inline ShippingTest.php]: In both tests, you use `$ship->method('isOrderShipped')->willReturn(true);`

**Pull request merged**